Add `getUsername` method to TotpBean.java to support otp:// URI scheme

summersab opened the pull request:
Implements #16917 

Add a `getUsername()` method to the `TotpBean` class so usernames can be used in the TOTP templates. For example, this will make it possible to add otp:// URIs to the `login-config-totp.ftl` template:

```
<a href="otpauth://totp/${(realm.displayName)?replace(" ", "%20")}:${(totp.username)}?secret=${(totp.totpSecretEncoded)?replace(" ", "")}&period=${(totp.policy["period"])}&algorithm=${(totp.policy["algorithm"])?replace("Hmac", "")}&digits=${(totp.policy["digits"])}">
    Add to OTP App
</a>
```

This is much more convenient for mobile users as opposed to copying over the manual TOTP strings into their OTP app. Since the username field is part of the otp:// URI scheme, having it available for the template is useful.

<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

pedroigor commented:
@summersab LGTM, thanks.

Could you please follow the guidelines here https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md?

Basically, rebase, squash commits, create an issue/enhancement, and link to this PR as per guideline above.

summersab committed:
Update TotpBean.java

Add a `getUsername()` method to the `TotpBean` class so usernames can be used in the TOTP templates.

summersab commented:
Was there something I missed, @pedroigor?

pedroigor commented:
@summersab All good, tks.
